hooks/useCreatePost.tsx

```diff
@@ -0,0 +1,100 @@
+import Router from "next/router";
+import * as React from "react";
+import { GameStreetAction, GameType } from "../models/GameSituation";
+import { PostTitle, PostBody } from "../models/Post";
+import { PostData, PostingGameSituation } from "../repositories/createPost";
+import useRepository from "./useRepository";
+
+export default function useCreatePost() {
```

Comment:
Several variables in this function are defined but used once. Mostly you can use just `if (x === y) { ... }` instead of `const isXY = x === y;  if (isXY) { ... }`